desktop/notification/notify.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package notification implements bindings to D-Bus notification
+// specification, version 1.2, as documented at https://developer.gnome.org/notification-spec/
+package notification
+
+import (
+	"fmt"
+)
+
+// Message describes a single notification message.
+//
+// The notification should be related to a specific application, as indicated by
+// AppName. In practice this should be the name of the desktop file and could be
+// also accompanied by an appropriate hint indicating which icon to use.
+//
+// Message must include a summary and should include a body. The body may use
+// HTML-like markup language to include bold, italic or underline text, as well
+// as to include images and hyperlinks.
+//
+// A notification can automatically expire after the given number of
+// milliseconds. This is separate from the notification being visible or
+// invisible on-screen. Expired notifications are removed from persistent
+// message roster, if one is supported.
+//
+// A notification may replace an existing notification by setting the ReplacesID
+// to a non-zero value. This only works if the notification server was not
+// re-started and should be used for as long as the sender process is alive, as
+// sending the identifier across session startup boundary has no chance of
+// working correctly.
+//
+// A notification may optionally carry a number of hints that further customize it
+// in a specific way. Refer to various hint constructors for details.
+//
+// A notification may optionally also carry one of several actions. If
+// supported, actions can be invoked by the user, broadcasting a notification
+// response back to the session. This mechanism only works if there is someone
+// listening for the action being triggered.
+//
+// In all cases, the specific notification must take into account the
+// capabilities of the server. For instance, if a server does not support body
+// markup, then such markup is not automatically stripped by either the client
+// or the server.
+type Message struct {
+	AppName       string
+	Icon          string
+	Summary       string
+	Body          string
+	ExpireTimeout int32 // XXX: consider using time.Duration instead?
```

Comment:
+1 for time.Duration, alternatively add a comment saying what is the unit, is it second or milliseconds?

Reply:
I'll go for a duration in a follow-up if you don't mind.